pkg/auth/providers/azure/clients/client.go

```diff
@@ -29,41 +25,15 @@ type AzureClient interface {
 
 // NewAzureClient returns a new client to be used for first-time authentication. This means it does not need any
 // externally stored secrets with tokens, as is the case when a client is created to use an existing access token.
-func NewAzureClient(useMSGraph bool) AzureClient {
-	if useMSGraph {
-		return &azureMSGraphClient{}
-	}
-	return &azureADGraphClient{}
-}
-
-// GetPrincipalID attempts to extract the ID of either a user or group from the principal value.
-func GetPrincipalID(principal v3.Principal) string {
-	name := principal.ObjectMeta.Name
-	if parts := strings.Split(name, "//"); len(parts) > 1 {
-		return parts[1]
-	}
-	return ""
-}
-
-// ExtractFieldFromJWT attempts to extract a value from a JWT by field name.
-// It does not make assumptions about the type of the return value, so the caller is responsible for handling that.
-func ExtractFieldFromJWT(tokenString string, fieldID string) (interface{}, error) {
-	pieces := strings.Split(tokenString, ".")
-	if len(pieces) != 3 {
-		return "", fmt.Errorf("invalid token")
-	}
-	decoded, err := base64.RawURLEncoding.DecodeString(pieces[1])
-	if err != nil {
-		return "", fmt.Errorf("error decoding token")
-	}
-
-	var data map[string]interface{}
-	err = json.Unmarshal(decoded, &data)
-	if err != nil {
-		return "", fmt.Errorf("error unmarshaling token")
-	}
-	if _, ok := data[fieldID]; !ok {
-		return "", fmt.Errorf("missing field %s from token", fieldID)
+func NewAzureClient(config *v32.AzureADConfig, credential *v32.AzureADLogin, useDeprecatedAzureADClient bool) (AzureClient, error) {
```

Comment:
A couple of things here:
1) The msgraph client is still not initialized - so anything using the new flow will still need to have the `setInternalFields` called, which is not preferable. 
2) This is using a non-exported function of the adGraphClient (configureServicePrinciaplToken) rather than using an exported Constructor, which seems odd to me. 

I would recommend the following:
1) Initialize the MSGraph client using an exported constructor.
2) Create a constructor for AzureADGraphClient, even if it's just a wrapper around configureServicePrincipalToken.

I would also like to note that we have a `newAzureClient` method in the provider - I'm kind of thinking that method and this one overlap in functionality but don't seem to know about each other. Could you elaborate on why we have these as separate methods.

Reply:
Resolving, change made.